word-search/TonyKim9401.java

```diff
@@ -0,0 +1,52 @@
+// TC: O(n * m * 4^k);
+// -> The size of board: n * m
+// -> Check 4 directions by the given word's length: 4^k
+// SC: O(n * m + k)
+// -> boolean 2D array: n * M
+// -> recursive max k spaces
+class Solution {
+    public boolean exist(char[][] board, String word) {
+        // Mark visited path to do not go back.
+        boolean[][] visit;
+
+        for (int i = 0; i < board.length; i++) {
+            for (int j = 0; j < board[0].length; j++) {
+                visit = new boolean[board.length][board[0].length];
```

Comment:
48번 째 줄에서 `visit[i][j]`를 다시 `true`로 복원을 해주시니, 굳이 재귀 함수를 호출할 때 마다 매번 새로운 집합을 생성해야할까 하는 생각이 들었습니다.

```suggestion
        boolean[][] visit = new boolean[board.length][board[0].length];

        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board[0].length; j++) {
```

Reply:
좋은 의견 감사합니다!
맞네요. 재귀적으로 반환시 true 였던 값이 false로 바뀌어 재귀가 끝날때 마다 복원될텐데 하나의 2차원 배열만 사용해도 괜찮을것 같습니다. 감사합니다!